core/src/test/scala/unit/kafka/admin/AclCommandTest.scala

```diff
@@ -130,22 +132,61 @@ class AclCommandTest extends ZooKeeperTestHarness with Logging {
   private def createServer(): Unit = {
     servers = Seq(TestUtils.createServer(KafkaConfig.fromProps(brokerProps)))
     val listenerName = ListenerName.forSecurityProtocol(SecurityProtocol.PLAINTEXT)
-    adminArgs = Array("--bootstrap-server", TestUtils.bootstrapServers(servers, listenerName))
+
+    val tmp = File.createTempFile(classOf[AclCommandTest].getName, "createServer")
+    tmp.deleteOnExit()
+    val pw = new PrintWriter(tmp)
+    pw.println("client.id=my-client")
+    pw.close()
+
+    adminArgs = Array("--bootstrap-server", TestUtils.bootstrapServers(servers, listenerName),
+      "--command-config", tmp.getAbsolutePath)
+  }
+
+  private def callMain(args: Array[String]): (String, String) = {
+    val appender = LogCaptureAppender.createAndRegister()
+    val previousLevel = LogCaptureAppender.setClassLoggerLevel(classOf[AppInfoParser], Level.WARN)
+    val outErr = TestUtils.grabConsoleOutputAndError(AclCommand.main(args))
+    LogCaptureAppender.setClassLoggerLevel(classOf[AppInfoParser], previousLevel)
+    LogCaptureAppender.unregister(appender)
+    Assert.assertEquals("Command should execute without logging errors or warnings",
+      "",
+      appender.getMessages.map{event => s"${event.getLevel} ${event.getMessage}" }.mkString("\n"))
+    outErr
   }
 
   private def testAclCli(cmdArgs: Array[String]): Unit = {
+
     for ((resources, resourceCmd) <- ResourceToCommand) {
       for (permissionType <- Set(ALLOW, DENY)) {
         val operationToCmd = ResourceToOperations(resources)
         val (acls, cmd) = getAclToCommand(permissionType, operationToCmd._1)
-          AclCommand.main(cmdArgs ++ cmd ++ resourceCmd ++ operationToCmd._2 :+ "--add")
-          for (resource <- resources) {
-            withAuthorizer() { authorizer =>
-              TestUtils.waitAndVerifyAcls(acls, authorizer, resource)
-            }
+        val (addOut, addErr) = callMain(cmdArgs ++ cmd ++ resourceCmd ++ operationToCmd._2 :+ "--add")
+        assertOutputContains("Adding ACLs", resources, resourceCmd, addOut)
```

Comment:
If not mistaken, `--add` also lists all the ACLs. Should we verify both here?

Instead of adding everything here (e.g. list in the middle), it may be easier to add a simpler test case which verifies the output when say 2 ACLs are added. We could do the same for `--list` and `--remove`. I am not sure if it is really better or not. It is just a suggestion.

Reply:
@dajac I'm not completely sure what you mean. You're suggesting we just test the output for, say, the cross product `(--topic foo, --group bar)` × `(--add, --list, --remove)` in a new test case?

It do think that currently it's difficult to construct the expected output correctly, and that makes the test harder to understand.

Reply:
@tombentley Yes, I meant something along these lines. I am not a huge fan of having one test that verifies everything so I wish that we could move towards having smaller and simpler test cases in the future. I do agree that this is outside of the scope of this PR. The other test case `testAclCliWithClientId` that you have added is good to cover the fix.